### ISSUE TITLE
.Deprecated() instead of warning()

### ISSUE THREAD

**Issue by HenrikBengtsson** [mschubert/clustermq#114]
Noticed that `clustermq::workers()` uses:
```r
warning("'log_worker' is deprecated, use template(log_file=...) instead")
```
to notify user about deprecated features.  I recommend `.Deprecated()`, especially since [one day we'll hopefully](https://github.com/HenrikBengtsson/Wishlist-for-R/issues/74) have that `R CMD check downstream.pkg_1.0.tar.gz` will report on deprecated usages.

For defunct functions, there's `.Defunct()` - instead of a plan `stop()` error.

**Comment by mschubert** [comment]
I'm not sure this fits here. The `.Deprecated` documentation states:

> When an object is about to be removed from R it is first deprecated and should include a call to ‘.Deprecated’.
> new: character string: A suggestion for a replacement function.

In any case, I'm considering again passing all template values to the worker, so it can use e.g. `withTimeout` to shut down cleanly before walltime instead of being killed.

**Comment by HenrikBengtsson** [comment]
Not sure I understand.  I basically propose:
```r
.Deprecated(msg = "'log_worker' is deprecated, use template(log_file=...) instead")
```
and it sounds like you are eventually going to stop supporting `log_worker` - that day, you'll switch the above to:
```r
.Defunct(msg = "'log_worker' is deprecated, use template(log_file=...) instead")
```
to generate an error.  Eventually, you'll completely drop it and `log_worker` will become an "unknown" argument.

**Comment by mschubert** [comment]
Ok, I should have tested this: I assumed the `new` argument is required, but it is not and `msg` suffices.

You're totally right.

**Comment by HenrikBengtsson** [comment]
Yeah, the neat is that in R 3.6.0 those warnings will be of class `deprecatedWarning`.  This is the first step for `R CMD check` to be able to distinguish those from regular warnings (which are not reported on).  So, hopefully, one day `R CMD check` will be able to give a WARNING when a package uses a deprecated function in another package.  Right now package developers of don't notice those until the function becomes defunct (=error).

**Comment by mschubert** [comment]
Yes, that part I got. Thank you!

**Comment by HenrikBengtsson** [comment]
BTW, this message should say:
```Q(..., log_file="/path/to.file")
'log_worker' is deprecated, use template = list(log_file=...) instead
```
instead of
```
'log_worker' is deprecated, use template(log_file=...) instead
```
correct?
